src/graph/entities/graph_entity.c

```diff
@@ -48,6 +49,14 @@ static void _GraphEntity_RemoveProperty(const GraphEntity *e, Attribute_ID attr_
 
 /* Add a new property to entity */
 SIValue *GraphEntity_AddProperty(GraphEntity *e, Attribute_ID attr_id, SIValue value) {
+	ASSERT(e);
+
+	// Emit an error and return if we're trying to add an invalid type.
+	if(!(SI_TYPE(value) & SI_PRIMITIVES)) {
```

Comment:
Catching the error at this point might be a bit too late
```
MATCH (l:L),(r:R),(q:Q) SET l.v = 1, r.v = 2, q.v = l
```
Might leave the graph in a none deterministic state, there isn't much we can do about it beside turn SET/UPDATE into an egger operation. I think the way to counter this is via a rollback functionality (MVCC)

Reply:
You're right, I've added this logic to OpCreate and OpUpdate to catch errors before anything is committed. This still is a risk for MERGE...ON MATCH/CREATE SET, but the surface area is at least diminished.